core/include/oneseismic/azure.hpp

```diff
@@ -15,7 +15,7 @@ inline constexpr const char* x_ms_version() noexcept (true) {
 
 class az : public one::storage_configuration {
 public:
-    az(std::string acc, std::string k);
+    az(std::string acc, std::string k, std::string base_url = "https://{}.blob.core.windows.net");
```

Comment:
This line is too long, consider breaking it up.

Reply:
More importantly, the fmtlib {} syntax shouldn't bleed into the argument like that. Don't pass arguments that then expects to be formatted in a public interface.

Instead, the base url should be completely static, and any configuration should be done by the caller.

Reply:
We do need to somehow inject the storage account into the URL since this is set per batch while the URL is static.

Reply:
Doesn't that just make whatever comes after {} static, and could be used instead? With implicit https://, which is not unlikely to change. Base url = blob.core.windows.net means problem solved, without format injection

Reply:
Not quite, if we are running against azurite the storage account comes after.

Reply:
Then azure should be considered a different storage type, with its own rules, shouldn't it? And it should definitely not be a default argument to a function a substitution pattern built in, that's just asking for trouble.